File: backend/app/schemas/docs.py
"""Pydantic schemas for Documentation API."""

from datetime import datetime
from pydantic import BaseModel


class DocBadgeResponse(BaseModel):
    """Status badge extracted from documentation."""

    label: str
    value: str
    status: str  # "passing", "failing", "warning"
    source_file: str
    line_number: int


class DocHeadingResponse(BaseModel):
    """Heading for table of contents."""

    level: int
    text: str
    id: str


class DocTreeNodeResponse(BaseModel):
    """Tree node for documentation navigation."""

    id: str
    title: str
    filename: str
    children: list["DocTreeNodeResponse"]
    modified_at: datetime


class DocContentResponse(BaseModel):
    """Full document content with metadata."""

    id: str
    title: str
    filename: str
    content: str
    badges: list[DocBadgeResponse]
    headings: list[DocHeadingResponse]
    modified_at: datetime


class DocStatusResponse(BaseModel):
    """Live status summary for polling."""

    badges: list[DocBadgeResponse]
    last_modified: datetime | None
    files_checked: int


class DocSearchResultResponse(BaseModel):
    """Search result item."""

    doc_id: str
    filename: str
    line: int
    match: str
    context: str


class ProjectPulseResponse(BaseModel):
    """AI-generated project health summary."""

    summary: str
    metrics: dict[str, str]
    citations: list[dict]
    generated_by: str  # "rule-based" or "claude"

File: frontend/types/docs.ts
export type DocBadge = {
  label: string;
  value: string;
  status: "passing" | "failing" | "warning";
  source_file: string;
  line_number: number;
};

export type DocHeading = {
  level: number;
  text: string;
  id: string;
};

export type DocTreeNode = {
  id: string;
  title: string;
  filename: string;
  children: DocTreeNode[];
  modified_at: string;
};

export type DocContent = {
  id: string;
  title: string;
  filename: string;
  content: string;
  badges: DocBadge[];
  headings: DocHeading[];
  modified_at: string;
};

export type DocStatus = {
  badges: DocBadge[];
  last_modified: string | null;
  files_checked: number;
};

export type DocSearchResult = {
  doc_id: string;
  filename: string;
  line: number;
  match: string;
  context: string;
};

export type ProjectPulse = {
  summary: string;
  metrics: Record<string, string>;
  citations: Array<{ file: string; line: number; excerpt: string }>;
  generated_by: "rule-based" | "claude";
};
